realm-jni/src/io_realm_internal_table.cpp

```diff
@@ -1500,3 +1505,33 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_Table_nativeSetPrimaryKey(
     } CATCH_STD()
     return 0;
 }
+
+// Fixes interop issue with Cocoa Realm where the Primary Key table had different types.
+// This affects:
+// - All Realms created by Cocoa and used by Realm-android up to 0.80.1
+// - All Realms created by Realm-Android 0.80.1 and below
+// See https://github.com/realm/realm-java/issues/1059
+JNIEXPORT void JNICALL Java_io_realm_internal_Table_nativeMigratePrimaryKeyTableIfNeeded
+    (JNIEnv* env, jobject, jlong groupNativePtr, jlong privateKeyTableNativePtr)
+{
+    Group* group = G(groupNativePtr);
+    Table* pk_table = TBL(privateKeyTableNativePtr);
+    if (pk_table->get_column_type(io_realm_internal_Table_PRIMARY_KEY_FIELD_COLUMN_INDEX) == type_Int) {
+        StringData tmp_col_name = StringData("tmp_field_name");
+        std::size_t tmp_col_ndx = pk_table->add_column(DataType(type_String), tmp_col_name);
```

Comment:
I think the core team is moving to a policy of of not importing the std namespace in the header. Should we do the same?